src/core/core.scale.js

```diff
@@ -668,16 +676,11 @@ module.exports = function(Chart) {
 						context.lineTo(itemToDraw.tx2, itemToDraw.ty2);
 					}
 
-					if (gridLines.drawOnChartArea) {
-						context.moveTo(itemToDraw.x1, itemToDraw.y1);
-						context.lineTo(itemToDraw.x2, itemToDraw.y2);
-					}
-
 					context.stroke();
 					context.restore();
 				}
 
-				if (optionTicks.display) {
```

Comment:
Might make sense to do this for any value of `itemToDraw.label` that isn't truthy. Blank labels don't need to be drawn and same with `null` labels. What do you think?